parser/src/parser/tests/list_comprehension.rs

```diff
@@ -1,437 +1,234 @@
-                Identifier("x".to_string()),
-                VariableValueExpr::ListComprehension(ListComprehension::new(
-                    Exp(
-                        Box::new(SymbolAccess(SymbolAccess::new(
-                            Identifier("col".to_string()),
-                            AccessType::Default,
-                            0,
-                        ))),
-                        Box::new(Const(7)),
-                    ),
-                    vec![(
-                        Identifier("col".to_string()),
-                        Iterable::Identifier(Identifier("c".to_string())),
-                    )],
-                )),
-            )),
-            BoundaryStmt::Constraint(BoundaryConstraint::new(
-                SymbolAccess::new(Identifier("a".to_string()), AccessType::Default, 0),
-                Boundary::First,
-                Add(
-                    Box::new(Add(
-                        Box::new(Add(
-                            Box::new(SymbolAccess(SymbolAccess::new(
-                                Identifier("x".to_string()),
-                                AccessType::Vector(0),
-                                0,
-                            ))),
-                            Box::new(SymbolAccess(SymbolAccess::new(
-                                Identifier("x".to_string()),
-                                AccessType::Vector(1),
-                                0,
-                            ))),
-                        )),
-                        Box::new(SymbolAccess(SymbolAccess::new(
-                            Identifier("x".to_string()),
-                            AccessType::Vector(2),
-                            0,
-                        ))),
-                    )),
-                    Box::new(SymbolAccess(SymbolAccess::new(
-                        Identifier("x".to_string()),
-                        AccessType::Vector(3),
-                        0,
-                    ))),
-                ),
-                None,
-            )),
-        ]),
-    ]);
-
-    ParseTest::new().expect_ast(source, expected);
+    let mut expected = Module::new(ModuleType::Root, SourceSpan::UNKNOWN, ident!(test));
+    expected
+        .trace_columns
+        .push(trace_segment!(0, "$main", [(a, 1), (b, 1), (c, 4)]));
+    expected.public_inputs.insert(
+        ident!(inputs),
+        PublicInput::new(SourceSpan::UNKNOWN, ident!(inputs), 2),
+    );
+    expected.integrity_constraints = Some(Span::new(
+        SourceSpan::UNKNOWN,
+        vec![enforce!(eq!(access!(a), int!(0)))],
+    ));
+    expected.boundary_constraints = Some(Span::new(
+        SourceSpan::UNKNOWN,
+        vec![
+            let_!(x = lc!(((col, expr!(access!(c)))) => exp!(access!(col), int!(7))).into() =>
+                  enforce!(eq!(bounded_access!(a, Boundary::First), add!(add!(add!(access!(x[0]), access!(x[1])), access!(x[2])), access!(x[3]))))),
```

Comment:
note: As mentioned previously, this shouldn't really be valid, since we can't access the "current" or "next" value of columns in boundary constraints. An integrity constraint formed this way would be valid (assuming it's enforced against `a` rather than `a.first`). Changing the list comprehension to something else that doesn't reference a trace column would make it valid, e.g. `let x = [i^7 for i in 0..4]`.